publiq-ui/pub-input.vue

```diff
@@ -0,0 +1,53 @@
+<template>
+  <div class="input-container">
+    <pub-label v-if="label" id="form-input">{{ label }}</pub-label>
```

Comment:
I think this needs a `for=` to link it to the input field (so you can click it to focus on the field, so screen readers know what field it's related to, ...)

Reply:
the id prop is passing it to for="" on the actual label, first I was passing a prop 'for' but there was complaining because of the naming

Reply:
I don't seem to be able to focus on the text field in the `pub-input` story by clicking on the label though.

Reply:
Ah, I think it's actually caused by what @jdrieghe mentioned, that we re-use the same id in multiple places now.
Because the `for` is indeed there in the HTML

<img width="732" alt="Screenshot 2020-09-15 at 14 36 44" src="https://user-images.githubusercontent.com/959026/93212463-eb043a80-f762-11ea-9fb9-14ae96165c40.png">

Reply:
I could just create a uuid to be sure, will need to add a new dependency though

Reply:
How about we add a `name` prop to the `pub-input` component and make that required? Then we can use that for the `id` and `for`? A `uuid` feels a bit... random 😄 But I can see where you're coming from, you probably want to avoid having to set some kind of id/name manually since we probably don't need to identify the field from the outside of the component?

Reply:
Yes, since the only link is to match the label with the input, it seemed appropriate  😉 But your idea is also a good option I think. There will be no way to be sure if the given name isn't already used elsewhere though.

Reply:
> There will be no way to be sure if the given name isn't already used elsewhere though.

True. Maybe for now, we can set an id based on the `this._uid` that exists in each component (but prefixed for example)? See https://stackoverflow.com/a/34957052 

Then we don't have to set it manually since we have no use case for that right now, but the "random" id is still a bit "predictable" in a sense. If we do need the ability to set a custom id later, we can always add that as a prop then?

Reply:
Cool to know that each component has access to its uid! 🤙  I'll do it using the component's idea then

Reply:
I think the uid is a good solution for now. I would be very careful with using a `name` property in the context of an input element. Those have an entirely different meaning in html.

Reply:
After @simon-debruijn changed it to use `this._uid` we read that it's not supposed to be used actually, it's considered a "private" property 😕 The alternative is a uuid, but that would introduce a dependency on another package/function. Maybe the `name` prop but with another name, like `id`? Makes more sense maybe as it would set that as the HTML id then.